discovery/service/kafka_broker.py

```diff
@@ -325,25 +327,28 @@ def _build_custom_listeners(self, service_prop: dict) -> tuple:
             ssl_enabled = service_prop.get(key2)
             if ssl_enabled is not None:
                 custom_listeners[name]['ssl_enabled'] = True
-
-            if 'ssl_mutual_auth_enabled' in self.inventory.groups.get('kafka_broker').vars and \
-                    self.inventory.groups.get('kafka_broker').vars.get('ssl_mutual_auth_enabled') is True:
-                custom_listeners[name]['ssl_mutual_auth_enabled'] = True
+                ssl_mutual_enabled = service_prop.get(key3)
+                logger.info(f"SSL mutual auth enabled for listener {name} {ssl_mutual_enabled}")
+                if ssl_mutual_enabled == 'required':
+                    custom_listeners[name]['ssl_mutual_auth_enabled'] = True
+                else:
+                    custom_listeners[name]['ssl_mutual_auth_enabled'] = False
 
             sasl_protocol = service_prop.get(key1)
             if sasl_protocol is not None:
-                custom_listeners[name]['sasl_protocol']: sasl_protocol
+                custom_listeners[name]['sasl_protocol'] = sasl_protocol
                 # Add the users to corresponding sasl mechanism
                 key = f"listener.name.{name.lower()}.{sasl_protocol.lower()}.sasl.jaas.config"
                 _dict = locals()[f"default_{sasl_protocol.lower().replace('-', '_')}_users"]
                 _dict.update(self.__get_user_dict(service_prop, key))
                 self.mapped_service_properties.add(key)
+            else:
+                custom_listeners[name]['sasl_protocol'] = 'none'
 
         return self.group, {
             "kafka_broker_custom_listeners": custom_listeners,
-            "sasl_scram_users": default_scram_users,
-            "sasl_scram256_users": default_scram256_users,
-            "sasl_scram512_users": default_scram_sha_512_users,
```

Comment:
Can you please explain the reason to remove scram512 users dict?

Reply:
cp-ansible has only sasl_scram_users or sasl_scram256_users, the former is used for SCRAM-SHA-512 so just wanted this to be more inline with what cp-ansible does.

Reply:
Discovery would be running against the cluster which is mostly probably not maintained by cp-ansible. If such cluster has scram512 users, what would be our strategy. Are we planning to drop them?

Reply:
Not dropping - we get the scram config from the sasl.mechanism which is either SCRAM-SHA-256 or SCRAM-SHA-512. Then - depending on which type it is we fill the users dict. To match cp-ansible we use the scram_users and scram265_users. We are only getting the scram users by parsing the sasl.jaas.config line anyway - the rest is in Zookeeper and we would need to use CLI - kafka-configs to get usernames from it (can't get passwords).